src/app/core/auth/directives/has-some-roles.directive.ts

```diff
@@ -14,12 +14,11 @@ import { SessionService } from '../session.service';
 export class HasSomeRolesDirective extends AbstractIfThenElseDirective implements OnInit {
 	roles: Array<string | Role>;
 	constructor(
-		viewContainer: ViewContainerRef,
 		templateRef: TemplateRef<any>,
 		private sessionService: SessionService,
```

Comment:
For all of these classes that extend from `AbstractIfThenElseDirective`, I choose not to use `inject` for `viewContainer`.  The reasoning was that I was unsure how it would interact in the case that the thenTemplate is overridden like in the example below.

```html
<div *hasSomeRoles="['user', 'admin']; then thenBlock"></div>
<ng-template #thenBlock>...</ng-template>
```

Reply:
In line with my last comment, while making this change I would advocate for just moving `sessionService` and `authorizationService` to injects rather than in the constructor, but again this is more of a nit pick.